Penumbra/Import/Models/Import/VertexAttribute.cs

```diff
@@ -121,89 +122,174 @@ public static VertexAttribute Position(Accessors accessors, IEnumerable<Accessor
+                        return byteValue > 0;
+                    if (adjustment > 0)
+                        return byteValue < 255;
+
+                    return true;
+                })
+                .Select(index => (index, delta: Math.Abs(originalValues[index] - convertedValues[index])))
+                .MinBy(x => x.delta)
+                .index;
+            byteValues[closestIndex] = (byte)(byteValues[closestIndex] + Math.CopySign(1, adjustment));
+            adjustment               = 255 - byteValues.Select(value => (int)value).Sum();
+        }
+
+        return byteValues;
     }
 
     public static VertexAttribute? BlendIndex(Accessors accessors, IDictionary<ushort, ushort>? boneMap, IoNotifier notifier)
     {
-        if (!accessors.TryGetValue("JOINTS_0", out var jointsAccessor))
+        if (!accessors.TryGetValue("JOINTS_0", out var joints0Accessor))
             return null;
 
-        if (!accessors.TryGetValue("WEIGHTS_0", out var weightsAccessor))
+        if (!accessors.TryGetValue("WEIGHTS_0", out var weights0Accessor))
             throw notifier.Exception("Mesh contained JOINTS_0 attribute but no corresponding WEIGHTS_0 attribute.");
 
         if (boneMap == null)
             throw notifier.Exception("Mesh contained JOINTS_0 attribute but no bone mapping was created.");
 
-        var element = new MdlStructs.VertexElement()
+        var joints0       = joints0Accessor.AsVector4Array();
+        var weights0      = weights0Accessor.AsVector4Array();
+        accessors.TryGetValue("JOINTS_1", out var joints1Accessor);
+        accessors.TryGetValue("WEIGHTS_1", out var weights1Accessor);
+        var element = new MdlStructs.VertexElement
         {
             Stream = 0,
-            Type   = (byte)MdlFile.VertexType.UByte4,
+            Type   = (byte)MdlFile.VertexType.UShort4,
             Usage  = (byte)MdlFile.VertexUsage.BlendIndices,
         };
 
-        var joints = jointsAccessor.AsVector4Array();
-        var weights = weightsAccessor.AsVector4Array();
+        var joints1  = joints1Accessor?.AsVector4Array();
+        var weights1 = weights1Accessor?.AsVector4Array();
 
         return new VertexAttribute(
             element,
             index =>
             {
-                var gltfIndices = joints[index];
-                var gltfWeights = weights[index]; 
+                var gltfIndices0 = joints0[index];
+                var gltfWeights0 = weights0[index];
+                var gltfIndices1 = joints1?[index];
+                var gltfWeights1 = weights1?[index];
+                var v0 = new Vector4(
+                    gltfWeights0.X == 0 ? 0 : boneMap[(ushort)gltfIndices0.X],
+                    gltfWeights0.Y == 0 ? 0 : boneMap[(ushort)gltfIndices0.Y],
+                    gltfWeights0.Z == 0 ? 0 : boneMap[(ushort)gltfIndices0.Z],
+                    gltfWeights0.W == 0 ? 0 : boneMap[(ushort)gltfIndices0.W]
+                );
+                
+                Vector4 v1;

```

Comment:
```suggestion
                var v1 = Vector4.Zero;

```
can omit the `else` case, then, i think